front/src/modules/trainschedule/components/ManageTrainSchedule/Map.tsx

```diff
@@ -79,24 +79,24 @@ const Map = ({ pathProperties, setMapCanvas }: MapProps) => {
 
   const mapRef = useRef<MapRef | null>(null);
 
-  const captureMap = async () => {
-    const mapElement = document.getElementById('map-container');
-    if (!mapElement) {
-      console.error('Map element not found');
-      return;
-    }
-
-    try {
-      const canvas = await html2canvas(mapElement);
-      const imageDataURL = canvas.toDataURL();
-      if (setMapCanvas) setMapCanvas(imageDataURL);
-    } catch (error) {
-      console.error('Error capturing map:', error);
-    }
-  };
-
   useEffect(() => {
-    captureMap();
+    const captureMap = async () => {
```

Comment:
why do you put the function declaration inside the useEffect ?

Reply:
Because it was asked in a previous PR : https://github.com/OpenRailAssociation/osrd/pull/7080#discussion_r1601257414

and because of the asynchronous operation, it's better to define the function within the useEffect